test/mongoDbTest.js

```diff
@@ -23,78 +23,181 @@ const genreArr = [
   'thriller',
   'western',
 ];
-const id = Math.floor(Math.random() * 300000);
+const id = Math.floor(Math.random() * 1000000);
 const genre = Math.floor(Math.random() * 15);
 
 describe('UserMovies', () => {
-  it('should contain at least 1,000,000 users', () => {
+  it('should contain at least 1,000,000 users', (done) => {
     mongoDb.UserMovies.count()
       .then((count) => {
         const isBigEnough = count >= 1000000;
         isBigEnough.should.equal(true);
       })
       .catch((err) => {
-        throw err;
+        done(err);
       });
+    done();
   });
 
-  it('each user should have a recs list', () => {
-    mongoDb.UserMovies.findById(id, (err, result) => {
-      if (err) {
-        throw err;
-      }
-      result.recs.should.be.a('object');
-    });
+  it('each user should have a recs list', (done) => {
+    mongoDb.UserMovies.findById(id)
+      .then((result) => {
+        result.recs.should.be.a('array');
+      })
+      .catch((err) => {
+        done(err);
+      });
+    done();
+  });
+
+  it('each rec movie should have a movieId', (done) => {
+    mongoDb.UserMovies.findById(id)
+      .then((result) => {
+        result.recs[0].movieId.should.be.a('number');
+      })
+      .catch((err) => {
+        done(err);
+      });
+    done();
+  });
+
+  it('each rec movie should have a title', (done) => {
```

Comment:
I'm noticing that your invocations of the done callback are not waiting for the promise chain to be resolved. You should be calling done() at the end of all asynchronous events (e.g. on line 68 instead of line 72).